src/booster.rs

```diff
@@ -200,6 +260,62 @@ impl Booster {
         Ok(output)
     }
 
+    /// Get number of evaluation metrics
+    pub fn num_eval(&self) -> Result<i32> {
+        let mut out_len = 0;
+        lgbm_call!(lightgbm_sys::LGBM_BoosterGetEvalCounts(
+            self.handle,
+            &mut out_len
+        ))?;
+        Ok(out_len)
+    }
+
+    /// Get names of evaluation metrics
+    pub fn eval_names(&self) -> Result<Vec<String>> {
+        let num_metrics = self.num_eval()?;
+        let metric_name_length = 32;
+        let mut num_eval_names = 0;
+        let mut out_buffer_len = 0;
+        let out_strs = (0..num_metrics)
+            .map(|_| {
+                CString::new(" ".repeat(metric_name_length))
+                    .unwrap()
+                    .into_raw() as *mut c_char
+            })
+            .collect::<Vec<_>>();
+        lgbm_call!(lightgbm_sys::LGBM_BoosterGetEvalNames(
+            self.handle,
+            num_metrics,
+            &mut num_eval_names,
+            metric_name_length as u64,
+            &mut out_buffer_len,
+            out_strs.as_ptr() as *mut *mut c_char
+        ))?;
+        let output: Vec<String> = out_strs
+            .into_iter()
+            .map(|s| unsafe { CString::from_raw(s).into_string().unwrap() })
+            .take(num_eval_names as usize)
+            .collect();
```

Comment:
I know the string logic is taken from `feature_name()`,j but if you are getting segfaults this (and feature_names) might be worth investigating. It is a bit suspicious, especially how the [CString::from_raw](https://doc.rust-lang.org/std/ffi/struct.CString.html#method.from_raw) documentation says you aren't supposed to change the string's length, which this probably does.

A better solution might be to allocate the strings as Vec<u8> initialized with 0s, and read them in with [CString::from_vec_with_nul](https://doc.rust-lang.org/std/ffi/struct.CString.html#method.from_vec_with_nul)

Reply:
Made a PR that should fix this, and makes the eval_names test pass: #11